### PR TITLE
add com.gitlab.maevemi.publictransport

### PR DESCRIPTION
**Pull request by maevemi**
### Please confirm your submission meets all the criteria

- [ x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [ x] My pull request follows the instructions at [App Submission][submission].
- [ x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [ x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ x] I am an upstream contributor to the project.
- [x ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: com.gitlab.maevemi.publictransport.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
+        ],
```
Comment (hfiguiere) [inline com.gitlab.maevemi.publictransport.json:21]: build-args for network acces are not allowed. You have to either generate a manifest of the cargo dependencies 
(see https://github.com/flatpak/flatpak-builder-tools) or use vendored crates.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by maevemi** [comment]
Thanks for the hints and sorry that I overlooked this point. I updated the manifest to use vendored crates.

**Comment by bilelmoussaoui** [comment]
bot, build com.gitlab.maevemi.publictransport

**Comment by bilelmoussaoui** [comment]
It seems like you forgot to unparent some widgets on dispose, nothing critical though
```
(publictransport:2): Gtk-WARNING **: 23:14:23.661: Finalizing GtkSearchEntry 0x562c6625ecc0, but it still has children left:

(publictransport:2): Gtk-WARNING **: 23:14:23.661:    - PublictransportLocationPopover 0x562c6626c560

(publictransport:2): Gtk-WARNING **: 23:14:23.662: Finalizing GtkButton 0x562c662964a0, but it still has children left:

(publictransport:2): Gtk-WARNING **: 23:14:23.662:    - ProductsUsed 0x562c66246260

(publictransport:2): Gtk-WARNING **: 23:14:23.662: Finalizing GtkSearchEntry 0x562c6625e980, but it still has children left:

(publictransport:2): Gtk-WARNING **: 23:14:23.662:    - PublictransportLocationPopover 0x562c6626c2e0
 ```

**Pull request closed without merging**